py/h2o_wave/h2o_wave/share.py

```diff
@@ -0,0 +1,26 @@
+import asyncio
+
+
+async def pipe(r: asyncio.StreamReader, w: asyncio.StreamWriter) -> None:
+    while True:
+        data = await r.read(4096)
+        if not data:
+            break
+        w.write(data)
+        await w.drain()
+
+
+async def listen_on_socket(local_host: str, local_port: int, remote_host: str, remote_port: int) -> None:
+    while True:
```

Comment:
Should this be an infinite cycle?

Reply:
I believe so. The client should expose the app until the process is stopped via CTRL+C. Note that Wave uses HTTP 1.1 as of today meaning once the request is done (e.g. CSS resource), the connection is closed, signaling the requestor there is no more data to come, thus a new connection needs to be made and added to the pool.

Reply:
Yes, you are right. I somewhat didn't realize this was a main loop.